File: DeviantArtFs/Types/ResponseTypes/DeviantArtMoreLikeThisPreviewResult.fs
namespace DeviantArtFs

open System
open FSharp.Json

type DeviantArtMoreLikeThisPreviewResult = {
    seed: Guid
    author: DeviantArtUser
    more_from_artist: Deviation list
    more_from_da: Deviation list
} with
    static member Parse json = Json.deserialize<DeviantArtMoreLikeThisPreviewResult> json
    member this.GetMoreFromArtist() = OptUtils.listDefault this.more_from_artist
    member this.GetMoreFromDA() = OptUtils.listDefault this.more_from_da

File: DeviantArtFs.Types/DeviantArtMoreLikeThisPreviewResult.cs
using System;

namespace DeviantArtFs.Types {
    public class DeviantArtMoreLikeThisPreviewResult {
        public Guid seed { get; set; }
        public DeviantArtUser author { get; set; } = new DeviantArtUser();
        public Deviation[] more_from_artist { get; set; } = new Deviation[0];
        public Deviation[] more_from_da { get; set; } = new Deviation[0];
    }
}
